src/base.jl

```diff
@@ -276,8 +304,14 @@ function add_component!(sys::System, component::T) where T <: Component
         throw(InvalidParameter("$(component.name) is already stored for type $T"))
     end
 
+    error_detected = false
+
+    if !isempty(sys.validation_descriptor)
+        error_detected = validate_fields(sys, component)
+    end
+
     sys.components[T][component.name] = component
-    return nothing
+    return error_detected
```

Comment:
I know why you did this (get all errors logged), but I'm not sure it is what we want.  This forces the user to check the return of the function to see if the component was added.  Our standard practice is to throw an exception on an error.

The log-all-validation errors logic applies to validating a whole system, not adding an individual component.  It's a subtle different but still important.  We should discuss.